scripts/nb-tester/qiskit_docs_notebook_tester/__init__.py

```diff
@@ -37,21 +38,64 @@
 matplotlib.set_loglevel("critical")
 """
 
-# If not submitting jobs, we also run this code before notebook execution to mock the real backend
-MOCKING_CODE = """\
+MOCKING_CODE = """
 import warnings
-from qiskit_ibm_runtime import QiskitRuntimeService
-from qiskit.providers.fake_provider import GenericBackendV2
-
-def patched_least_busy(self, *args, **kwarg):
-  return GenericBackendV2(num_qubits=6, control_flow=True)
-
-QiskitRuntimeService.least_busy = patched_least_busy
 
 warnings.filterwarnings("ignore", message="Options {.*} have no effect in local testing mode.")
 warnings.filterwarnings("ignore", message="Session is not supported in local testing mode or when using a simulator.")
+warnings.filterwarnings("ignore", message="Session is not supported in local testing mode or when using a simulator.")
 """
 
+def generate_backend_patch(
+    backend_name, 
+    provider: Literal["qiskit_ibm_runtime", "qiskit_fake_provider", "runtime_fake_provider"] = "qiskit_ibm_runtime", 
+    **kwargs
```

Comment:
Rather than using `kwargs`, which is vague/very generic, you can call this `runtime_service_args: dict[str, str | None]`. It will be more clear and keeps `**kwargs` available for future changes to this API.